packages/remix/src/hydrogen.ts

```diff
@@ -0,0 +1,110 @@
+import { basename } from 'path';
+import { Node, Project } from 'ts-morph';
+
+/**
+ * For Hydrogen v2, the `server.ts` file exports a signature like:
+ *
+ * ```
+ * export default {
+ *   async fetch(
+ *     request: Request,
+ *     env: Env,
+ *     executionContext: ExecutionContext,
+ *   ): Promise<Response>;
+ * }
+ * ```
+ *
+ * Here we parse the AST of that file so that we can:
+ *
+ *  1. Convert the signature to be compatible with Vercel Edge functions
+ *     (i.e. `export default (res: Response): Promise<Response>`).
+ *
+ *  2. Track usages of the `env` parameter which (which gets removed),
+ *     so that we can create that object based on `process.env`.
+ */
+export function patchHydrogenServer(
+  project: Project,
+  serverEntryPoint: string
+) {
+  const sourceFile = project.addSourceFileAtPath(serverEntryPoint);
+  const defaultExportSymbol = sourceFile.getDefaultExportSymbol();
```

Comment:
In my tests, `defaultExportSymbol` is `undefined` but my `server.js` has:

```
export default {
  async fetch(request, env, executionContext) {
```

Is that to be expected?

Reply:
Nice catch. Should be fixed by https://github.com/vercel/vercel/pull/10305/commits/a68723dd49d8b16a5fefb95353bfdfeb377c8bb9. I'll follow-up and add an E2E test as well.